src/finalizer/index.ts

```diff
@@ -136,17 +147,26 @@ export async function finalize(
         // @dev Likely to be the 2nd part of a 2-stage withdrawal (i.e. retrieve() on the Polygon bridge adapter).
         message = "Unknown finalizer simulation failure.";
       }
-      logger.info({ at: "finalizer", message, reason, txn: _txn });
+      logger.warn({ at: "finalizer", message, reason, txn: _txn });
       return false;
     }
   });
 
   if (finalizations.length > 0) {
-    let txn: TransactionReceipt;
+    let txn: TransactionResponse;
     try {
-      // Note: If the sum of finalizations approaches the gas limit, consider slicing them up.
-      const txns = finalizations.map(({ txn }) => txn);
-      txn = await (await multicall2.aggregate(txns)).wait();
+      const finalizerTxns = finalizations.map(({ txn }) => txn);
+      const txnToSubmit: AugmentedTransaction = {
+        contract: multicall2,
+        chainId: hubChainId,
+        method: "aggregate",
+        args: finalizerTxns,
+        gasLimit: gasEstimation,
+        gasLimitMultiplier: 2,
+        message: `Batch finalized ${finalizerTxns.length} withdrawals and/or proofs`,
+        mrkdwn: `Batch finalized ${finalizerTxns.length} withdrawals and/or proofs`,
+      };
+      txn = (await txnClient.submit(hubChainId, [txnToSubmit]))[0];
```

Comment:
nit: I think this is a little easier to read
```suggestion
      [txn] = await txnClient.submit(hubChainId, [txnToSubmit]);
```